tests/layer_validation_tests.h

```diff
@@ -270,6 +270,24 @@ class VkLayerTest : public VkRenderFramework {
         return GetPhysicalDeviceProperties2(props2);
     }
 
+    template <typename Proc>
+    const Proc GetInstanceProcAddr(const char *proc_name, bool assert_proc = true) {
+        auto proc = reinterpret_cast<Proc>(vk::GetInstanceProcAddr(instance(), proc_name));
+        if (assert_proc) {
+            assert(proc);
+        }
+        return proc;
+    }
```

Comment:
Instead of having `assert_proc` be a runtime variable, what about a template parameter?

```suggestion
    template <typename Proc, bool assert_proc = true>
    const Proc GetInstanceProcAddr(const char *proc_name) {
        auto proc = reinterpret_cast<Proc>(vk::GetInstanceProcAddr(instance(), proc_name));
        if constexpr (assert_proc) {
            assert(proc);
        }
        return proc;
    }
```

Reply:
Yep this is better!